### PR TITLE
Merge master jdk-11.0.21+1 into openj9-staging

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Merge master `jdk-11.0.21+1` into `openj9-staging`

Resolved conflicts:
```
	src/java.base/share/classes/com/sun/crypto/provider/SunJCE.java
	src/java.base/share/classes/java/security/Security.java
	src/java.base/share/classes/sun/security/provider/SunEntries.java
	src/java.base/share/classes/sun/security/rsa/RSAPrivateCrtKeyImpl.java
	src/java.base/share/classes/sun/security/rsa/RSAPublicKeyImpl.java
	src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/SunPKCS11.java
	src/jdk.crypto.ec/share/classes/sun/security/ec/SunEC.java
	test/jdk/java/lang/StringBuilder/HugeCapacity.java
```

Modified:
```
	jdk.crypto.ec/share/classes/sun/security/ec/SunEC.java
	jdk.crypto.ec/share/classes/sun/security/ec/NativeECUtil.java
	jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java
```

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeECUtil.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ static String getCurveName(ECParameterSpec params) {
         String curveName;
         if (params instanceof NamedCurve) {
             NamedCurve namedCurve = (NamedCurve) params;
-            curveName = namedCurve.getName();
+            curveName = namedCurve.toString();
```
Comment (keithc-ca) [inline closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeECUtil.java:110]: This doesn't yield the name of the curve; this should be
```java
            curveName = namedCurve.getNameAndAliases()[0];
```
(see the [head stream](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/closed/src/java.base/share/classes/sun/security/ec/NativeECUtil.java#L109)).

File: src/java.base/share/classes/com/sun/crypto/provider/SunJCE.java [hunk 2]
```diff
@@ -302,300 +263,261 @@ void putEntries() {
...
+                        attrs);
         } else {
                 ps("Cipher",  "ChaCha20",
                         "com.sun.crypto.provider.ChaCha20Cipher$ChaCha20Only",
                         null, attrs);
-                ps("Cipher",  "ChaCha20-Poly1305",
+                psA("Cipher",  "ChaCha20-Poly1305",
                         "com.sun.crypto.provider.ChaCha20Cipher$ChaCha20Poly1305",
-                        chachaPolyAliases, attrs);
+                        attrs);
```
Comment (keithc-ca) [inline src/java.base/share/classes/com/sun/crypto/provider/SunJCE.java:295]: While you're here, please correct the indentation of these lines (it's four spaces too much).

File: src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java [hunk 3]
```diff
@@ -412,7 +416,7 @@ static PrivateKey privateKey(Session session, long keyID, String algorithm,
         if (keySensitive && (SunPKCS11.mysunpkcs11 != null) && "RSA".equals(algorithm)) {
             try {
                 byte[] key = SunPKCS11.mysunpkcs11.exportKey(session.id(), attributes, keyID);
-                RSAPrivateKey rsaPrivKey = RSAPrivateCrtKeyImpl.newKey(key);
+                RSAPrivateKey rsaPrivKey = RSAPrivateCrtKeyImpl.newKey(KeyType.RSA, "PKCS#1", key);
```
Comment (keithc-ca) [inline src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java:419]: The second argument must be `"PKCS#8"` (see the [head stream](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java#L437)).

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeECUtil.java (hunk 1 above)
Comment (JasonFengJ9) [reply closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeECUtil.java]: Fixed

File: src/java.base/share/classes/com/sun/crypto/provider/SunJCE.java (hunk 2 above)
Comment (JasonFengJ9) [reply src/java.base/share/classes/com/sun/crypto/provider/SunJCE.java]: Removed

File: src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java (hunk 3 above)
Comment (JasonFengJ9) [reply src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java]: Fixed

**Comment by keithc-ca** [comment]
Upstream has adopted a newer version of harfbuzz. I expect AIX builds will fail unless the rest of https://github.com/ibmruntimes/openj9-openjdk-jdk21/pull/24 is applied here.

Concretely, in `hb-map.hh` the definitions of `keys()` and `values()` near line 348 should be
```c
  auto keys () const HB_AUTO_RETURN
  (
    + keys_ref ()
    | hb_map (hb_ridentity)
  )
```
and
```c
  auto values () const HB_AUTO_RETURN
  (
    + values_ref ()
    | hb_map (hb_ridentity)
  )
```

**Comment by JasonFengJ9** [comment]
Passed [an internal build](https://hyc-runtimes-jenkins.swg-devops.com/job/Build_JDK11_ppc64_aix_Personal/1077/console)

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**